solr/solr-ref-guide/modules/upgrade-notes/pages/major-changes-in-solr-9.adoc

```diff
@@ -71,6 +71,13 @@ Due to changes in Lucene 9, that isn't possible any more.
 === Configuration
 In solrconfig.xml, the `numVersionBuckets` and `versionBucketLockTimeoutMs` settings are now obsolete and ignored; a warning will be logged if specified.
 
+=== Partial Results
+When solr is used for reporting, legal discovery or other use cases that cannot accept partial results, yet it still needs to be protected from long-running queries, users may now opt out of receiving partial results.
+This feature should eliminate some wasted processing and bandwidth, as well as simplifying client side logic, and reducing the chance of inappropriate responses to the user.
+This can be enabled either via a request parameter (`allowPartialResults`), environment variable or system property.
+The partialResults attribute in the request header will now have three possible states: `"true"`, `"omitted"` or absent, with the new `"omitted"` value indicating the difference between a timed out request with results omitted due to query limits and results that are merely empty because the query didn't match any docs.
```

Comment:
Somehow this sentence is very confusing to me ... maybe this: the new "omitted" value indicating that query returned partial results, which due to the setting were not allowed and they were then discarded? or something more English-like ;)

Reply:
Yeah probably needs to be broken into 2 sentences at least.

Reply:
Hmm, maybe using "discarded" instead of "omitted", both in the docs AND in the response header, would make it clearer?